components/butterfly/src/rumor.rs

```diff
@@ -117,13 +154,22 @@ impl fmt::Display for RumorKey {
 
 /// A representation of a Rumor; implemented by all the concrete types we share as rumors. The
 /// exception is the Membership rumor, since it's not actually a rumor in the same vein.
-pub trait Rumor: Message<ProtoRumor> + Sized {
+pub trait Rumor: Message<ProtoRumor> + Sized + fmt::Debug + fmt::Display {
     fn kind(&self) -> RumorType;
     fn key(&self) -> &str;
     fn id(&self) -> &str;
     fn merge(&mut self, other: Self) -> bool;
 }
 
+pub trait Transient: Rumor {
```

Comment:
I can try this and see how it goes, but off the cuff, I'm guessing it's going to cause problems.  A lot of the code in its current state works because `RumorStore` can be holding either `Rumor`s or `Transient`s.  I might be off base on this though, so I'm happy to try it out and see how it goes.

Reply:
RumorStore is declared as `struct RumorStore<T>` so it can hold *any* type. However, some `impl` blocks have trait bounds, so that means those particular methods will only be available on types that implement those bounds. If you run into roadblocks with this approach, let me know.